dotnet/src/Connectors/Connectors.OpenAI/Core/OpenAIFunction.cs

```diff
@@ -129,47 +142,55 @@ internal OpenAIFunction(
     /// <returns>A <see cref="ChatTool"/> containing all the function information.</returns>
     public ChatTool ToFunctionDefinition()
     {
-        BinaryData resultParameters = s_zeroFunctionParametersSchema;
+        BinaryData resultParameters = this.Strict ? s_zeroFunctionParametersSchema_strict : s_zeroFunctionParametersSchema;
 
         IReadOnlyList<OpenAIFunctionParameter>? parameters = this.Parameters;
         if (parameters is { Count: > 0 })
         {
             var properties = new Dictionary<string, KernelJsonSchema>();
             var required = new List<string>();
 
-            for (int i = 0; i < parameters.Count; i++)
+            foreach (var parameter in parameters)
             {
-                var parameter = parameters[i];
-                properties.Add(parameter.Name, parameter.Schema ?? GetDefaultSchemaForTypelessParameter(parameter.Description));
-                if (parameter.IsRequired)
+                properties.Add(parameter.Name, parameter.Schema ?? this.GetDefaultSchemaForTypelessParameter(parameter.Description));
+                if (parameter.IsRequired || this.Strict)
```

Comment:
set to null when not initially required https://platform.openai.com/docs/guides/structured-outputs#all-fields-must-be-required